src/pages/address/_components/contract/ReadWriteContract.tsx

```diff
@@ -18,14 +22,43 @@ export default function ReadWriteContract({
   addressHash: string;
   implementationAddress: string | null;
 }) {
+  const [isLoading, setIsLoading] = useState(true);
   const [expandAll, setExpandAll] = useState<boolean>(false);
   const [resetForm, setResetForm] = useState<boolean>(false);
   const { connection } = useNetwork();
-  const { data: methods, isLoading } = useGetContractMethodsQuery({
-    network: connection,
-    addressHash,
-    type,
-  });
+  const [trigger] = useGetContractMethodsMutation();
+  const [methods, setMethods] = useState<SmartContractMethod[]>([]);
+
+  const fetchMethods = async () => {
+    setIsLoading(true);
+    const data = await trigger({
+      network: connection,
+      addressHash,
+      type,
+    }).unwrap();
+
+    if (
+      type === ContractMethodType.Read ||
+      type === ContractMethodType.ReadProxy
+    ) {
+      // Filter out read methods that are also in write methods
```

Comment:
The comment is misleading, let's delete/rephrase it since we changed the tech approach